sphinxcontrib/spelling/checker.py

```diff
@@ -5,8 +5,12 @@
 """Spelling checker extension for Sphinx.
 """
 
-import enchant
-from enchant.tokenize import get_tokenizer
+try:
+    import enchant
+    from enchant.tokenize import get_tokenizer
+    have_enchant = True
+except ImportError:
+    have_enchant = False
```

Comment:
Usually, it's better to have less things (ideally just one instruction) in the `try` section:
```suggestion
except ImportError as imp_exc:
    enchant_import_error = imp_exc
else:
    enchant_import_error = None
```
Also, preserving the import error may help convey more details later.